src/coreclr/tools/superpmi/CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
+if (CLR_CMAKE_HOST_UNIX)
+    add_compile_options(-Wno-format)
```

Comment:
Don't mean to unblock this PR, but how many violations are there?

Previously we fixed violations and removed `Wno-format` from a few places, and the upcoming libunwind 1.7 was going to remove its last occurrence in the repo, not anymore. 🥲

Reply:
@am11 Absolutely agree on this point. There are a lot unfortunately - If I recall, the number was in the hundreds :-/. Many of them are around the confusion between `%ll*` and `%l*`. The issue is on Windows `ll` is what we want for `*int64_t`, but on non-Windows we want `l` because it reduces to `long`, not `long long`.

Reply:
I've also filed https://github.com/dotnet/runtime/issues/81316, which should help keep everyone honest in the future. We can also look to address these and other violations in that work item. As always, any help in this area is most appreciated.

Reply:
> Many of them are around the confusion between `%ll*` and `%l*`.

This sounds familiar. Mono also defines some of such formats per platform in cmake config: https://github.com/dotnet/runtime/blob/844f6eff7f22b654e413c2c5491b1445f0d1ee13/src/mono/CMakeLists.txt#L774-L791 (we fixed a few warnings to get the count down to 0 and deleted Wno-format IIRC)

We can move these defines in `src/native/minipal/utils.h` (or a separate header file in that directory) which is shared between all native code in the repo.